### PR TITLE
Ref #1499: Add link preview to long press context menu.

### PR DESCRIPTION
**Pull request by iccub**
## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Screenshots:

Zrzut ekranu 2019-10-10 o 18 56 27

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -2511,11 +2511,21 @@ extension BrowserViewController: WKUIDelegate {
             return UIMenu(title: url.absoluteString, children: actions)
         }
         
-        let config = UIContextMenuConfiguration(identifier: nil, previewProvider: nil, actionProvider: actionProvider)
+        let linkPreview: UIContextMenuContentPreviewProvider = {
+            return SFSafariViewController(url: url)
```
Comment (kylehickinson) [inline Client/Frontend/Browser/BrowserViewController.swift]: Is it possible to make our own preview provider that uses BVC in some way (so shields are loaded in and such)

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Also, Safari provides a way to disable these previews, we may want to give the same option to users

**Comment by kylehickinson** [comment]
Sorry that review meant to be `comment`, not `request changes`

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController.swift]: UIContextMenuContentPreviewProvider accepts any `UIViewController` so we could try to add some custom view with content blocker lists

**Comment by iccub** [comment]
I think I should add a preference to hide preview links, similar to what Safari does at the moment

**Comment by iccub** [comment]
Added content blocking to the link preview and an option to hide link previews

Zrzut ekranu 2019-10-11 o 14 53 41
Zrzut ekranu 2019-10-11 o 14 53 29

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Somehow after disabling link previews, I no longer got the enable link previews option… It seems like something is supposed to be there (you can see the menu has a final separator down there but no text/image.

IMG_C03DD67E96AA-1

**Comment by kylehickinson** [comment]
Ahhh, never mind, that list scrolls, and wasn't visible because of the super long URL… Not sure if we should truncate it?

**Comment by iccub** [comment]
Added a 100char truncate to it, Safari truncates it too btw

Zrzut ekranu 2019-10-11 o 15 59 53

**Review by kylehickinson — APPROVED** [review]
Works for me

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**